### ISSUE TITLE
set returned by `discover()` can mutate

### ISSUE THREAD

**Issue by leanto** [SoCo/SoCo#397]
Recently, my program failed with the following error:

``` none
Traceback (most recent call last):
  File "./sonos-button.py", line 85, in <module>
    main()
  File "./sonos-button.py", line 52, in main
    coordinators = find_coordinators(zones)
  File "./sonos-button.py", line 79, in find_coordinators
    for zone in zones:
RuntimeError: Set changed size during iteration
```

This appears to have happened because `soco.discovery.discover()` does not return a static set, but rather [returns a reference to a set managed by a zone object](https://github.com/SoCo/SoCo/blob/22e787dad105ddb2814dfd4cd1d4d76e2349dc4a/soco/discovery.py#L177). This means that when the zone changes, the set's contents can change as well. And if the set's contents change in a separate thread due to an event, the above error can happen.

My suggestion would be to either change the documentation to make it clear that the returned set is dynamic and should be copied before iteration, or return a copy of the zone's set. (I have a slight preference for the documentation change since having a dynamic set is handy.)

**Comment by KennethNielsen** [comment]
Thanks for reporting this and for the analysis and sorry about the delay getting back to you.

I think I tend to lean towards returning a copy, since I can't see a good use case for a changing zone set. If one needs to watch for changes in the topology, it is properly better to watch the events instead.

**Comment by ghcs27** [comment]
I'm in favor of the copy as well.

**Comment by KennethNielsen** [comment]
I've added the "Byte size" label, which indicates that it is a good bug for new contributors to start with. Spread the word.

**Comment by KennethNielsen** [comment]
Closed by #582
